### DISCUSSION TITLE
Inquiry about Airfoil Optimization Constraints

### DISCUSSION THREAD

**Discussion by MachAeroDeveloper** [mdolab/dafoam#580] [General]
Dear friedenhe,
I am currently working on airfoil optimization using the provided DAFoam script, and I have a few inquiries regarding the constraints set in the optimization process.

1. **Constraint Parameters:**
   I noticed that in the script, the constraints for shape, thickness, and volume are set with lower and upper bounds. Specifically, for the shape constraint, the bounds are set as lower = -1 and upper = 1. I am curious to know if these values are constant for all types of airfoils, or if they are calculated dynamically based on certain considerations.

   Similarly, for the thickness and volume constraints, I would like to understand if the lower and upper bounds (-1 and 1, respectively) are universal for various airfoil simulations, or if there is a calculation involved that tailors these values to specific cases.

2. **Adjusting Lift Constraint Range:**
   In the optimization process, I am interested in adjusting the range for the lift constraint. Could you please guide me on how I can modify the lower and upper bounds for the lift constraint to better suit my specific requirements?

3. **Extracting Optimized Airfoil Profile:**
   Lastly, after the optimization process is complete, I would like to know how I can extract the newly optimized airfoil profile. Could you provide insights or steps on accessing and saving the optimized geometry for further analysis?

I appreciate your assistance and guidance on these matters. Your expertise will be invaluable in ensuring a successful and tailored airfoil optimization process.

Thank you for your time, and I look forward to your response.

Best regards,
Jamal Ali

**Comment by friedenhe** [comment]
1. For shape variables, upper and lower bounds are in meters. You can give large values such that the optimized shape variables do not hit the bound. The bounds should have no impact on the optimization results, no matter what case you are running. Please check the DAFoam workshop slides/videos on how the thickness/volume constraints work,, they should be generalable to other airfoils.
2. Not sure what you are trying to do, Please be more specific on "adjusting the range of the lift constraint", and why you want to do that.
3. Please check DAFoam website FAQ.

**Reply by MachAeroDeveloper** [reply]
Dear Professor Friedenhe,

I hope this message finds you well.

I wanted to express my sincere gratitude for your prompt response and clarification regarding the recent discussions we had. Your quick reply has been immensely helpful in guiding me through the challenges I encountered.

Regarding the first part of our discussion concerning the range, I fully understand that it will not significantly affect the solution. I am prepared to provide values within the specified range as needed.

In regards to the lift constraint, I have carefully considered your insights. Upon reflection, I recognize that while my objective function is focused on the drag coefficient (cd), the inverse of lift coefficient (1/cl) can substantially influence the lift, resulting in values that exceed acceptable limits. As discussed, I plan to incorporate a range for the drag constraint, such as 0.005 to 0.02, to mitigate the undesirable increase in lift, which can reach up to 400%. This approach aligns with the requirements for wind turbine applications and ensures that the solution remains within acceptable parameters.

Additionally, I intend to provide visual representation of the results, including figures generated from the Mach-Aero runscript, to further illustrate the impact of the constraints on the lift coefficient.

Once again, thank you for your invaluable guidance and support. Your expertise and timely assistance have been instrumental in advancing my understanding of the subject matter.

Looking forward to your continued guidance and collaboration.

Warm regards,
Ajlan
![Uploading image.png…]()
![Uploading Capture.JPG…]()

**Comment by friedenhe** [comment]
It seems that you want to minimize the drag. And you also want to control the lift to be within a range. It is doable in DAFoam, just set lower and upper bounds for the lift constraint.

**Reply by MachAeroDeveloper** [reply]
Dear Professor,friedenhe

Thank you for your guidance. Following your advice, I've successfully set a range for the constraints, which in this case are inequality constraints. I'm now curious about how the optimizer algorithm will operate when I apply these range constraints to the lift constraint.

My main concern is to ensure that applying these constraints doesn't lead to any errors or failures in the optimization process, particularly avoiding any issues related to inequality constraint errors. I'm also interested in understanding how these constraints might influence the resultant shape of the airfoil.

Any insights or recommendations you could provide on this matter would be greatly appreciated. Your expertise has been invaluable in helping me navigate these complex aspects of our project.

Thank you once again for your assistance.

Best regards,
Ajlan

**Comment by friedenhe** [comment]
Unfortunately, we provide support only for how to use DAFoam. We won't be able to provide suggestions on how to best set up your optimization..
